### PR TITLE
Testing / 23.0.4

### LINKED ISSUE #473: Unable to access settings
I use YunoHost on a computer at home as a server, after installing Nextcloud, impossible to access the config-panel of the application from the administrator interface. I got an error message regarding: Control panel versions '0.1' are not supported.

YunoHost version is: 4.3.6.2 (stable)
Nextcloud version is: 22.2.3~ynh1

Capture d’écran (4)

### PR DESCRIPTION
**Pull request by alexAubin**
## Changes

- https://github.com/YunoHost-Apps/nextcloud_ynh/pull/460
    - (this implies php 8.0)
- Upgrade config panel

Closes #473 

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by alexAubin** [comment]
!testme

**Review by ericgaspar — DISMISSED** [review]

File: scripts/install [hunk 1]
```diff
@@ -48,6 +49,7 @@ ynh_app_setting_set --app=$app --key=domain --value=$domain
 ynh_app_setting_set --app=$app --key=path --value=$path_url
 ynh_app_setting_set --app=$app --key=admin --value=$admin
 ynh_app_setting_set --app=$app --key=user_home --value=$user_home
+ynh_app_setting_set --app=$app --key=phpversion --value=$phpversion
```
Comment (kay0u) [inline scripts/install:52]: No need to do that:

https://github.com/YunoHost/yunohost/blob/dev/data/helpers.d/php#L84
https://github.com/YunoHost/yunohost/blob/d1ab1f674eb25f0970ded4698a89aee18f202cec/data/helpers.d/php#L120

**Review by kay0u — COMMENTED** [review]
LGTM anyway, I think we should manage the config panel before merging, and maybe wait a bit to get a nextcloud 23.0.1?

File: scripts/install (hunk 1 above)
Comment (ericgaspar) [reply scripts/install]: > and maybe wait a bit to get a nextcloud 23.0.1?

I have clearly no clues on how to setup config-panels... 😅

Comment (kay0u) [reply scripts/install]: I will try to spend some time on it :+1:

**Comment by kay0u** [comment]
^ tested on my side

**Comment by ericgaspar** [comment]
also tested and it works nicely. Bravo!

File: scripts/config [hunk 2]
```diff
@@ -9,90 +9,87 @@
...
+    then
+        ynh_app_setting_set --app=$app --key=fpm_footprint --value="$fpm_footprint"
+    fi
+}
 
-	ynh_return "YNH_CONFIG_MAIN_PHP_FPM_CONFIG_FOOTPRINT=$fpm_footprint"
-	ynh_return "YNH_CONFIG_MAIN_PHP_FPM_CONFIG_FREE_FOOTPRINT=$free_footprint"
-	ynh_return "YNH_CONFIG_MAIN_PHP_FPM_CONFIG_USAGE=$fpm_usage"
+set__free_footprint() {
+    if [ "$fpm_footprint" = "specific" ]
```
Comment (Tagadda) [inline scripts/config:60]:
```suggestion
    if [ "$fpm_footprint" == "specific" ]
```

File: scripts/config [hunk 3]
```diff
@@ -9,90 +9,87 @@
...
 
-    #=================================================
-    # RECONFIGURE PHP-FPM
-    #=================================================
-
-    if [ "$fpm_usage" != "$old_fpm_usage" ] || [ "$fpm_footprint" != "$old_fpm_footprint" ] || [ "$free_footprint" != "$old_free_footprint" ]
-    then
+    if [ "${changed[fpm_usage]}" == "true" ] || [ "${changed[fpm_footprint]}" == "true" ] || [ "${changed[free_footprint]}" == "true" ]; then
         # If fpm_footprint is set to 'specific', use $free_footprint value.
         if [ "$fpm_footprint" = "specific" ]
```
Comment (Tagadda) [inline scripts/config:75]:
```suggestion
        if [ "$fpm_footprint" == "specific" ]
```

**Review by Tagadda — CHANGES_REQUESTED** [review]
Some typos

**Comment by yalh76** [comment]
!testme

**Comment by tituspijean** [comment]
```
118578 WARNING Package php8.0-json is a virtual package provided by:
118581 WARNING   php8.0-phpdbg 8.0.15-1+0~20220129.29+debian10~1.gbpc7cb1f
118583 WARNING   php8.0-fpm 8.0.15-1+0~20220129.29+debian10~1.gbpc7cb1f
118584 WARNING   php8.0-cli 8.0.15-1+0~20220129.29+debian10~1.gbpc7cb1f
118586 WARNING   php8.0-cgi 8.0.15-1+0~20220129.29+debian10~1.gbpc7cb1f
118587 WARNING   libphp8.0-embed 8.0.15-1+0~20220129.29+debian10~1.gbpc7cb1f
118589 WARNING   libapache2-mod-php8.0 8.0.15-1+0~20220129.29+debian10~1.gbpc7cb1f
118594 WARNING E: Package 'php8.0-json' has no installation candidate
```

https://php.watch/versions/8.0/ext-json

It looks like this dependency is not needed anymore.

File: scripts/_common.sh [hunk 4]
```diff
@@ -1,9 +1,10 @@
+#!/bin/bash
 
 #=================================================
 # COMMON VARIABLES
 #=================================================
 
-YNH_PHP_VERSION="7.3"
+YNH_PHP_VERSION="8.0"
 pkg_dependencies="imagemagick libmagickcore-6.q16-6-extra acl tar smbclient at php${YNH_PHP_VERSION}-bz2 php${YNH_PHP_VERSION}-imap php${YNH_PHP_VERSION}-gmp php${YNH_PHP_VERSION}-gd php${YNH_PHP_VERSION}-json php${YNH_PHP_VERSION}-intl php${YNH_PHP_VERSION}-curl php${YNH_PHP_VERSION}-apcu php${YNH_PHP_VERSION}-redis php${YNH_PHP_VERSION}-ldap php${YNH_PHP_VERSION}-imagick php${YNH_PHP_VERSION}-zip php${YNH_PHP_VERSION}-mbstring php${YNH_PHP_VERSION}-xml php${YNH_PHP_VERSION}-mysql php${YNH_PHP_VERSION}-igbinary php${YNH_PHP_VERSION}-bcmath"
```
Comment (tituspijean) [inline scripts/_common.sh:8]:
```suggestion
pkg_dependencies="imagemagick libmagickcore-6.q16-6-extra acl tar smbclient at php${YNH_PHP_VERSION}-bz2 php${YNH_PHP_VERSION}-imap php${YNH_PHP_VERSION}-gmp php${YNH_PHP_VERSION}-gd php${YNH_PHP_VERSION}-intl php${YNH_PHP_VERSION}-curl php${YNH_PHP_VERSION}-apcu php${YNH_PHP_VERSION}-redis php${YNH_PHP_VERSION}-ldap php${YNH_PHP_VERSION}-imagick php${YNH_PHP_VERSION}-zip php${YNH_PHP_VERSION}-mbstring php${YNH_PHP_VERSION}-xml php${YNH_PHP_VERSION}-mysql php${YNH_PHP_VERSION}-igbinary php${YNH_PHP_VERSION}-bcmath"
```

Works in a manual test of the failing upgrade above (YunoHost 11).

**Comment by tituspijean** [comment]
!gogogadgetotitus

**Comment by tituspijean** [comment]
(ﾉ≧∇≦)ﾉ ﾐ ┸━┸
Here comes the "unmet dependencies" madness.

**Comment by ericgaspar** [comment]
!testme

**Comment by kay0u** [comment]
works on bullseye

**Comment by ericgaspar** [comment]
I haven't had any issue upgrading Nextcloud from previous version either. Maybe we should advertise the testing in the forum for user to test.

**Comment by kay0u** [comment]
Are you on Bullseye too?

**Comment by ericgaspar** [comment]
> Are you on Bullseye too?

yes I am

**Comment by kay0u** [comment]
Ok then, let's move the manifest to allow only yunohost >= 11.0

**Comment by zamentur** [comment]
Note: Nextcloud 23 comes with an option 'allow_local_remote_servers' useful to make onlyoffice working again (cause actually i suspect that all ynh setup with nc22 and onlyoffice have this issue : https://help.nextcloud.com/t/nc23-onlyoffice-host-violates-local-access-rules/128811 ).
The problem is due to dnsmasq that lie and give internal 127.0.0.1 ip instead of public ip for onlyoffice domains.

**Comment by Gredin67** [comment]
I tested this upgrade on Yunohost 11 successfully. I then installed the testing branch of onlyoffice https://github.com/YunoHost-Apps/onlyoffice_ynh/pull/72 
The bug with onlyoffice is not there anymore.

**Comment by ericgaspar** [comment]
!bullseyetestme

**Review by Tagadda — DISMISSED** [review]
LGTM 🚀

**Comment by ericgaspar** [comment]
This is to be released after Bullseye is released.

**Pull request closed without merging**